src/handle.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2019 Tyler Neely
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/// Provides access to underlying handles for 
+/// database operations
+pub trait Handle<T> {
```

Comment:
@iSynaptic Could you please provide a more information about this trait ? I couldn't get a goal of the trait.

Reply:
It allows traits to be automatically implemented for structs that expose certain handles. For example, both `DB` and `ReadOnlyDB` implement `Handle<ffi::rocksdb_t>` and `ops::Read`. Because of that, they automatically get an implementation (the same implementation) of `Get<'a>` and `GetCF<'a>` and other traits in the future. It will also allow `TransactionDB` and `OptimisticTransactionDB` to use the same implementation, once they are implemented.

Put in other words, any struct that exposes the appropriate handle and marker trait (`Read` or `Write`) will benefit from blanket trait implementations.